.github/workflows/test.yml

```diff
@@ -19,4 +19,48 @@ jobs:
       uses: actions/checkout@v2
 
     - name: Run Tests
-      run: go test ./...
+      run: go test ./packages/...
+
+  integration-test:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Check out code
+      uses: actions/checkout@v2
+
+    - name: Build GoShimmer Docker network
+      run: docker-compose -f tools/integration-tests/docker-compose.yml up -d --scale peer_replica=5
+
+    - name: Dispay containers
+      run: docker ps -a
+
+    - name: Run integration tests
+      run: docker-compose -f tools/integration-tests/tester/docker-compose.yml up --exit-code-from tester
+
+    - name: Stop GoShimmer Docker network
+      if: always()
```

Comment:
This technically means that this is also run even when already the checkout failed, but I am not sure there is a way around this